### DISCUSSION TITLE
AsgiFastStream not working with multiple workers?

### DISCUSSION THREAD

**Discussion by KingofGnome** [ag2ai/faststream#1946] [Q&A]
Hey,
i got a shost question on the asgi + multiple workers.
I'm running FastStream workers to consume a queue of long running tasks and is just awesome to work with the framework 💯 !

I've closely followed the [ASGI Support]( https://faststream.airt.ai/latest/getting-started/asgi/)-Tutorial to enable K8S probes and added Prometheus metrics just for the fun of it.

However, after using `FastStream(...).as_asgi(...)` like [here](https://faststream.airt.ai/latest/getting-started/asgi/#faststream-object-reuse) and running the app exactly like the tooltip suggests:

```
faststream run main:app --host 0.0.0.0 --port 8000 --workers 4
```
it seems to now ignore the --workers -flag and running a single process only.
Am i missing something or is it supposed to work like this?

Having long running low-cpu but io-bound tasks i'm really looking foreward to having multiple workers (on top of k8s scaling).
Right now the only approach i can see, is getting rid of the asgi, (bye bye grafana dashboard 😂) and switching to exec-based livenessprobes.

I'm trying to keep the project FastStream-"centered" and using asgi only as by-product for metrics and probes. An alternative would probably be to embed Faststream compleatly into Fastapi and run FastAPI/Uvicorn on multiple workers?

However, the docs kinda suggests that `as_asgi(...)` together with  `faststream run ... --workers` should work?
So either there is a solution i'm missing or we could improve the docs  :)

Thanks in advance.

**Comment by Sehat1137** [comment]
Hi, try to use something from this list for running your FastStream App
```shell
gunicorn -k uvicorn.workers.UvicornWorker main:app --workers=4
granian --interface asgi main:app --workers 4
hypercorn main:app --workers 4
```
We will fix this problem as soon as possible

**Comment by Lancetnik** [comment]
@KingofGnome yes, `faststream run main:asgi --workers X` should works. But, we are using **uvicorn** right now to run the app. And **uvicorn** doesn't use multithreading / multiprocessing to scale your app, so lifespan runs only once and our **FastStream** app runs in a single node as well. It is already described in #1874 Issue and I plan to merge #1936 PR with the fix today. Please, wait a bit for fix patch.

**Comment by Lancetnik** [comment]
@KingofGnome you can try 0.5.31 FastStream release - the bug was fixed there
